lib/AST/ASTPrinter.cpp

```diff
@@ -2585,10 +2588,8 @@ void PrintAST::printOneParameter(const ParamDecl *param, bool Curried,
   }
 
   // Set and restore in-parameter-position printing of types
-  auto prior = Options.PrintAsInParamType;
-  Options.PrintAsInParamType = true;
+  llvm::SaveAndRestore<bool> savePrintParam(Options.PrintAsInParamType, true);
   printTypeLoc(TheTypeLoc);
-  Options.PrintAsInParamType = prior;
```

Comment:
Yeah, it won't change anything today, but I agree it's worth it to improve local reasoning.  Will update.